### PR TITLE
feat(taskframework): add job scheduler to taskframework

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-feature
module-taskframework
#### What this PR does / why we need it:
1.integrate scheduler with task-framework, we can put a job describe by `JobDefinition` to `JobScheduler` to schedule,
and job will be scheduled by trigger config

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: script/start-odc.sh [hunk 1]
```diff
@@ -66,11 +66,13 @@ function check_env_value_set() {
 function init_parameters() {
     echo "init parameters start"
 
-    # init parameters require environment variable
-    check_env_value_set DATABASE_HOST "${DATABASE_HOST}"
-    check_env_value_set DATABASE_PORT "${DATABASE_PORT}"
-    check_env_value_set DATABASE_USERNAME "${DATABASE_USERNAME}"
-    check_env_value_set DATABASE_NAME "${DATABASE_NAME}"
+    if [ -z "${ODC_BOOT_MODE}"  ] || [ "${ODC_BOOT_MODE}" != 'TASK_EXECUTOR' ]; then
```
Comment (yhilmare) [inline script/start-odc.sh:69]: I think we should set these env event though we run odc as executor mode cause we may rely on metadb

Comment (krihy) [reply script/start-odc.sh:69]: yes , ODC_BOOT_MODE will be passed by  k8s pod environment, check_env_value_set  will be skipped when odc start by ODC_BOOT_MODE

Comment (krihy) [reply script/start-odc.sh]: revert this change

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Pull request merged**